### PR TITLE
Conditionals simple programs q8

### PR DESCRIPTION
**Pull request by jae186**
## PR Checklist
- [ ]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package conditionals.simple_programs_with_decision_points.hard.q8;
+/*
+Write a program that asks the user for two numbers, x and y then state if the sum of the two numbers is less or more than 50.
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java:3]:
```suggestion
Write a program that asks the user for two numbers, x and y then use a switch statement and the Integer.compare method to state if the sum of the two numbers is less or more than 50.
```

File: src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Enter x  
+1  
+Enter y  
+10  
+Sum is less than 50.
+
+ */
+import java.util.Scanner;
+
+public class sumToFifty {
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java:14]: capitalize sum

File: src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+
+        System.out.println("Enter x ");
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java:18]: add colons for consistency

File: src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            new NewLine(),
+            new StringLiteral("Enter y "),
+            new Optional(new StringLiteral(" ")),
+            new NewLine(),
+            new PlaceHolder()
+        };
+    }
+    
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java:36]: should add cases for 49 and 51

File: src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            // Case 1: 1 + 10 = 11, which is less than 50.
+            Arguments.of(1, 10, "Sum is less than 50"),
+            // Case 2: 25 + 25 = 50, exactly 50.
+            Arguments.of(25, 25, "Sum is exactly 50"),
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java:40]: I would recommend using a threshold variable in the maintest to make website variation easier

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java (hunk 2 above)
Comment (jae186) [reply src/conditionals/simple_programs_with_decision_points/hard/q8/sumToFifty.java:14]: done

**Comment by jae186** [comment]
This is ready to be reviewed again! Thanks for the feedback @SamamthaH  :)

File: src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    static final int THRESHOLD = 50;
+
+    @Override
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "1\n2";
+
+        return new Clause[]{
+            new StringLiteral("Enter x: "),
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java:27]: if this is Java, get rid of the extra spaces here because that's what the optionals are for, if it's parsons you don't need the optionals

File: src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            new PlaceHolder()
+        };
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+            // Case 1: Sum is less than the threshold
+            Arguments.of(1, 10, "Sum is less than " + THRESHOLD),
+            // Case 2: Sum is less than the threshold
+            Arguments.of(25, 24, "Sum is less than " + THRESHOLD),
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java:42]: you don't have to change this if you don't want to but an example for how a threshold variable is typically used would be instead of having 25 and 24, have THRESHOLD/2 and THRESHOLD/2 - 1 so when you change up the number of the threshold on the website you don't have to add additional variables for 25, 24, and 26

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+            // Case 1: Sum is less than the threshold
+            Arguments.of(1, THRESHOLD / 5, "Sum is less than " + THRESHOLD),
+            // Case 2: Sum is less than the threshold
+            Arguments.of(THRESHOLD / 2, THRESHOLD / 2 - 1, "Sum is less than " + THRESHOLD),
+            // Case 3: Sum is exactly the threshold
+            Arguments.of(25, 25, "Sum is exactly " + THRESHOLD),
```
Comment (SamamthaH) [inline src/conditionals/simple_programs_with_decision_points/hard/q8/MainTest.java:42]:
```suggestion
            Arguments.of(THRESHOLD / 2, THRESHOLD / 2, "Sum is exactly " + THRESHOLD),
```

**Review by SamamthaH — APPROVED** [review]
